clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3224,10 +3262,12 @@ namespace {
       }
       else {
         // Get the function prototype, where the abstract function return
-        // bounds are kept. The callee is always a function pointer.
+        // bounds are kept.  The callee (if it exists) 
+        // is always a function pointer.
         const PointerType *PtrTy =
           CE->getCallee()->getType()->getAs<PointerType>();
-        assert(PtrTy != nullptr);
+        if (PtrTy == nullptr)
```

Comment:
Leaving this assert in causes the Clang Parser/block-pointer-decl.c test to fail.